corehq/apps/users/views/__init__.py

```diff
@@ -701,25 +701,13 @@ def delete_domain_permission_mirror(request, domain, mirror):
 @require_POST
 def create_domain_permission_mirror(request, domain):
     form = CreateDomainPermissionsMirrorForm(domain=request.domain, data=request.POST)
-    if form.is_valid():
-        mirror_domain_name = form.cleaned_data.get("mirror_domain")
-        mirror_domain = Domain.get_by_name(mirror_domain_name)
-        if mirror_domain is not None:
-            try:
-                mirror = DomainPermissionsMirror(source=domain, mirror=mirror_domain_name)
-                mirror.save()
-            except IntegrityError:
-                message = _('"{mirror_domain_name}" has already been added.')
-                messages.error(request, message.format(mirror_domain_name=mirror_domain_name))
-            else:
-                message = _('You have successfully added the project space "{mirror_domain_name}".')
-                messages.success(request, message.format(mirror_domain_name=mirror_domain_name))
-        else:
-            message = _('Please enter a valid project space.')
-            messages.error(request, message.format())
-    else:
+    if not form.is_valid():
         for field, message in form.errors.items():
             messages.error(request, message)
```

Comment:
I didn't notice this earlier - looks like this isn't a standard CRUD operation where it re-renders the form with errors, it instead logs them as messages, so my previous comment about getting in-line errors was wrong.  Ah well.